File: python/runCmdCommand.py
import subprocess,os
from pathlib import Path

pythonPath = "python"
folders = ["","bin" , "python"]
def execute_microservice(a,b, microName):
    microPath = find(microName);
    if(not microPath): return "could'nt find " + microName
    if(a == None): a=""
    if(b == None): b=""
    command = pythonPath + " " + '"' + microPath + '"' + " " + str(a) + " " + str(b)
    return run_command(command)

def find(microName):
    if(".py" not in microName): microName+=".py"
    for folder in folders:
        microPath = folder+"\\"+microName
        if(Path(microPath).is_file()):
            return microPath
    x = os.path.dirname(os.path.abspath(__file__))
    for folder in folders:
        microPath = x+"\\" + folder + ("\\" if folder !="" else "") +microName
        if(Path(microPath).is_file()):
            return microPath
    return None

def run_command(command):
    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    (out,err) = p.communicate()
    if err : return err
    return out

File: javascript/runCmdCommand.js
const util = require("util");
const path = require('path')
const fs = require("fs")
const exec = util.promisify(require('child_process').exec);

let js = "node"
let folders = ["" , "bin" , "python" , "javascript"]
var execute_microservice = async function (a,b, microName){
    let microPath = find(microName);
    if(!microPath){return "could'nt find " + microName}
    if(a == null){a=""}
    if(b == null){b=""}
    let command = js + " " + '"' + microPath + '"' + " " + a + " " + b
    return run_command(command)
}
var find = function(microName){
    if(!(microName.includes(".js"))){microName+=".js"}
    let microPath = ""
    for (let folder of folders){
        microPath = folder + ((folder !="") ? "\\" : "") + microName
        if(fs.existsSync(microPath)){
            return microPath
        }
    }
    return null
}

var run_command = async function (command) {
  const { stdout, stderr } = await exec(command);
  console.log(stdout);
  return stdout
}

module.exports = {
    execute_microservice: execute_microservice
}

execute_microservice(5, 3, "minus.js")
